frontend/ui/widget/inputtext.lua

```diff
@@ -634,6 +639,30 @@ function InputText:onKeyPress(key)
     else
         handled = false
     end
+    if key["ScreenKB"] or key["Shift"] then
+        if key["Back"] then
+            self:delChar()
+        elseif key["Left"] then
+            self:leftChar()
+        elseif key["Right"] then
+            self:rightChar()
+        elseif key["Up"] then
+            self:upLine()
+        elseif key["Down"] then
+            self:downLine()
+        elseif key["Home"] then
+            if self.keyboard:isVisible() then
+                self:onCloseKeyboard()
+            else
+                self:onShowKeyboard()
+            end
+        elseif key["."] then
+            if Device:hasSymKey() then
+                self:addChars(",")
+                return true
```

Comment:
It is not because it works that it should go in - my suggestion was just for testing and seeing if there was a "later".
Now, you should make it cleaner - similar to the existing code you inserted that bit in the middle.
```lua
if not handled and (key["ScreenKB"] or key["Shift"]) then
    handled = true
    if .... then
    elseif ... then
    else
         handled = false
    end
end
if not handled and Device:hasDPad() then
```

Reply:
how about now?

Reply:
it's missing my parens in `x and (y or z)`.